### PR TITLE
Gradle Updates

### PR DESCRIPTION
**Pull request by Syer10**
Update Gradle and the dependancies.
Cleanup Gradle files, deduplicate code and dependency locations

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -27,26 +28,44 @@ val projects = listOf(
...
+
     dependencies {
         // Kotlin
         implementation(kotlin("stdlib-jdk8"))
         implementation(kotlin("reflect"))
         testImplementation(kotlin("test-junit5"))
 
         // coroutines
-        val coroutinesVersion = "1.5.1"
+        val coroutinesVersion = "1.5.2"
```
Comment (AriaMoradi) [inline build.gradle.kts:68]: Isn't coroutines version locked by kotlin?

File: build.gradle.kts [hunk 2]
```diff
@@ -55,22 +74,21 @@ configure(projects) {
...
+        implementation("io.github.microutils:kotlin-logging:2.0.11")
 
         // ReactiveX
         implementation("io.reactivex:rxjava:1.3.8")
         implementation("io.reactivex:rxkotlin:1.0.0")
         implementation("com.jakewharton.rxrelay:rxrelay:1.2.0")
 
         // dependency both in AndroidCompat and extensions, version locked by Tachiyomi app/extensions
-        implementation("org.jsoup:jsoup:1.14.1")
+        implementation("org.jsoup:jsoup:1.14.2")
```
Comment (AriaMoradi) [inline build.gradle.kts:91]: This is version locked by extensions, is it fine to use a newer version?

File: server/build.gradle.kts [hunk 3]
```diff
@@ -57,8 +43,7 @@ dependencies {
     implementation("com.github.inorichi.injekt:injekt-core:65b0440")
     implementation("com.squareup.okhttp3:okhttp:4.9.1")
     implementation("io.reactivex:rxjava:1.3.8")
-    implementation("org.jsoup:jsoup:1.14.1")
-    implementation("com.google.code.gson:gson:2.8.7")
+    implementation("com.google.code.gson:gson:2.8.8")
```
Comment (AriaMoradi) [inline server/build.gradle.kts:47]: same for gson, don't remove jsoup from here, it's for reference...

File: server/build.gradle.kts [hunk 4]
```diff
@@ -187,21 +160,14 @@ tasks {
...
         overwrite(shouldOverwrite())
     }
-
-    withType<LintTask> {
-        source(files("src/kotlin"))
-    }
-
-    withType<FormatTask> {
-        source(files("src/kotlin"))
-    }
```
Comment (AriaMoradi) [inline server/build.gradle.kts:206]: eh, dose it still work without config?

File: server/build.gradle.kts [hunk 5]
```diff
@@ -131,29 +116,17 @@ tasks {
...
-    withType<KotlinCompile> {
-        kotlinOptions {
-            freeCompilerArgs = listOf(
-                "-Xopt-in=kotlin.RequiresOptIn",
-                "-Xopt-in=kotlinx.coroutines.ExperimentalCoroutinesApi",
-                "-Xopt-in=kotlinx.coroutines.InternalCoroutinesApi",
-                "-Xopt-in=kotlinx.serialization.ExperimentalSerializationApi",
-            )
-        }
-    }
```
Comment (AriaMoradi) [inline server/build.gradle.kts:156]: we don't need this anymore?

File: build.gradle.kts (hunk 1 above)
Comment (Syer10) [reply build.gradle.kts:68]: It isnt usually, as long as you are not a major version behind you are good to update it as you like.

File: build.gradle.kts (hunk 2 above)
Comment (Syer10) [reply build.gradle.kts:91]: I believe Tachi has updated this in the app, so it should be fine

File: server/build.gradle.kts (hunk 4 above)
Comment (Syer10) [reply server/build.gradle.kts:206]: It was moved to the top-level build.gradle.kts

File: server/build.gradle.kts (hunk 5 above)
Comment (Syer10) [reply server/build.gradle.kts:156]: It was moved to the top-level build.gradle.kts

File: server/build.gradle.kts (hunk 3 above)
Comment (Syer10) [reply server/build.gradle.kts:47]: Duplicating it isn't a good idea, mismathing versions and all that. Maybe once we move to version catalogues or buildSrc version management I can re-add it?

Comment (AriaMoradi) [reply server/build.gradle.kts:47]: just don't do it for now...
I promise to be careful to not mismath.

Comment (Syer10) [reply server/build.gradle.kts:47]: 👌

**Pull request merged**